internal/pkg/service/cli/cmd/remote/table/download.go

```diff
@@ -0,0 +1,100 @@
+package table
+
+import (
+	"time"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/spf13/cobra"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/helpmsg"
+	common "github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/file/download"
+	"github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/table/unload"
+)
+
+func DownloadCommand(p dependencies.Provider) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   `download [table]`,
+		Short: helpmsg.Read(`remote/table/download/short`),
+		Long:  helpmsg.Read(`remote/table/download/long`),
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) (cmdErr error) {
+			// Ask for host and token if needed
+			baseDeps := p.BaseDependencies()
+			if err := baseDeps.Dialogs().AskHostAndToken(baseDeps); err != nil {
+				return err
+			}
+
+			// Get dependencies
+			d, err := p.DependenciesForRemoteCommand(common.WithoutMasterToken())
+			if err != nil {
+				return err
+			}
+
+			// Send cmd successful/failed event
+			defer d.EventSender().SendCmdEvent(d.CommandCtx(), time.Now(), &cmdErr, "remote-table-unload")
+
+			var tableID keboola.TableID
+			if len(args) == 0 {
+				allTables, err := d.KeboolaProjectAPI().ListTablesRequest(keboola.WithColumns()).Send(d.CommandCtx())
+				if err != nil {
+					return err
+				}
+
+				table, err := d.Dialogs().AskTable(d.Options(), *allTables)
+				if err != nil {
+					return err
+				}
+				tableID = table.ID
+			} else {
+				id, err := keboola.ParseTableID(args[0])
+				if err != nil {
+					return err
+				}
+				tableID = id
+			}
+
+			fileOutput, err := baseDeps.Dialogs().AskFileOutput(baseDeps.Options())
+			if err != nil {
+				return err
+			}
+
+			unloadOpts, err := parseUnloadOptions(d.Options(), tableID)
+			if err != nil {
+				return err
+			}
+
+			unloadedFile, err := unload.Run(d.CommandCtx(), unloadOpts, d)
+			if err != nil {
+				return err
+			}
+
+			fileWithCredentials, err := d.KeboolaProjectAPI().GetFileWithCredentialsRequest(unloadedFile.ID).Send(d.CommandCtx())
+			if err != nil {
+				return err
+			}
+
+			downloadOpts := download.Options{
+				File:          fileWithCredentials,
+				Output:        fileOutput,
+				ForceUnsliced: true,
```

Comment:
No, hele, to spojení slajs do jednoho souboru je fajn, ale asi není dobrý že jeden command to (bez ptaní) dělá a druhý (tj. file download) to nedělá. imho by to bylo dobrý sjednotit ať to dělají buď oba nebo žádný (a nebo případně i podle flagu který by byl defaultně true kdyby se ti hodně chtělo.)

Reply:
Máš pravdu že je to lepší, souhlasím, akorát jde teda o to, že bys to měl upravit i v tom file download ať to v jednom commandu nefunguje jinak než ve druhým (a ideálně by to snad mohlo sdílet stejnej kód).